tests/test-dladdr.c

```diff
@@ -0,0 +1,192 @@
+ * @param libname librray to get the address from
+ * @param addrsym symbol to get the address for
+ * @param should_match result should match the given values
+ * @return 0 check passed
+ * @return 1 check failed
+ */
+int check_dladdr_by_dlopen( char *libname, char *sym, int should_match )
+{
+    void *addr = get_symbol_address( libname, sym);
+    if (!addr)
+        return 1;
+    return check_dladdr( addr, sym, should_match );
+}
+
+#ifdef _WIN32
+#include <windows.h>
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+/* hide warning "reclared without 'dllimport' attribute" */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
+#elif defined(_MSC_VER)
+/* disable warning C4273 inconsistent dll linkage */
+#pragma warning(push)
+#pragma warning(disable: 4273)
+#endif
+/*
+ * Windows API functions decorated with WINBASEAPI are imported from iat.
+ * For testing force linking by import thunk for the following functions
+ */
+HMODULE WINAPI GetModuleHandleA (LPCSTR lpModuleName);
+SIZE_T WINAPI VirtualQuery (LPCVOID lpAddress, PMEMORY_BASIC_INFORMATION lpBuffer, SIZE_T dwLength);
+/* force linking by iat */
+/* WINBASEAPI, which is normally used here, can be overriden by compiler or application, so we cannot use it */
+__declspec(dllimport) HMODULE WINAPI LoadLibraryExA(LPCSTR lpLibFileName, HANDLE hFile, DWORD  dwFlags);
+__declspec(dllimport) int __cdecl atoi(const char *_Str);
+
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#pragma GCC diagnostic pop
+#elif defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+#endif
+
+#ifdef _WIN32
+#define PassWithoutSymbolOnWin PassWithoutSymbol
+#define PassWithDifferentAddressOnWin PassWithDifferentAddress
+#else
+#define PassWithoutSymbolOnWin Pass
+#define PassWithDifferentAddressOnWin Pass
+#endif
+
+#if defined(_WIN32) && !defined(DLFCN_WIN32_SHARED)
+#define PassWithSymbolOnSharedBuild PassWithoutSymbol
+#else
+#define PassWithSymbolOnSharedBuild Pass
+#endif
+
+#define UNUSED(x) (void)(x)
+
+int main(int argc, char **argv)
+{
+    UNUSED(argv);
+
+    int  result = 0;
+    if (argc == 2)
+        verbose = 1;
+
+    /* null pointer */
+    result = check_dladdr((void*)0, "null" , NoInfo);
+    /* invalid pointer */
+    result |= check_dladdr((void*)0x125, "invalid pointer" , NoInfo);
+    /* -ldl */
+    result |= check_dladdr( ((void*)dladdr), "dladdr" , PassWithSymbolOnSharedBuild );
+    /* -ldl */
+    result |= check_dladdr( (void*)dlopen, "dlopen", PassWithSymbolOnSharedBuild );
+    /* -lglibc */
+    result |= check_dladdr( (void*)atoi, "atoi", Pass );
+    /* test-dladdr */
+    result |= check_dladdr( (void*)main, "main", PassWithoutSymbolOnWin );
+    /* offsets */
+    result |= check_dladdr( (char*)dladdr-6, "dladdr-6", Fail );
+    /* offsets */
+    result |= check_dladdr( (char*)dladdr+6, "dladdr", PassWithDifferentAddress );
```

Comment:
In Visual Studio 2012 is this test failing also for dynamic build:
https://ci.appveyor.com/project/dlfcn-win32/dlfcn-win32/builds/36924363/job/93kcvg5s6ubbe7ls
```
check address 74DB1015 which has symbol 'dladdr' -> failed (filename: C:\projects\dlfcn-win32\build\bin\Debug\dl.dll base: 74DB0000 symbol name: 'dlsym' addr: 74DB1014)
```
It looks like that at address `dladdr+6` is different function (`dlsym`). Can you change this test from `dladdr+6` to `dladdr+1`. I hope that `+1` is small enough...